src/common/plugin.ts

```diff
@@ -47,7 +57,42 @@ export abstract class Plugin implements IPlugin {
 
     const response = this.container.channelService.hasPermission(channelName, this.permission);
     if (!response) {
-      message.reply(`Please use this command in a \`${this.permission}\` channel`);
+      const rooms = Object.values(Constants.Channels[this.permission]);
+      let addonText = '';
+
+      // not sure if this will cause issues as I do pop id, please let me know
+      const id = rooms
+        .filter((room) => {
+          return this.container.guildService
+            .getChannel(room)
+            .permissionsFor(message.member || '')
+            ?.has('VIEW_CHANNEL');
+        })
+        .map((room) => this.container.guildService.getChannel(room).id);
+
+      try {
+        if (this.permission.toString() === 'Private') {
+          addonText = ` This is primarily the class channels, and any channels we haven't defined.`;
+        } else if (id.length === 0) {
+          addonText = ' There are no permanent channels of this type.';
+        } else if (id.length === 1) {
+          addonText = ` <#${id[0]}> is the only channel whith this type.`;
+        } else if (id.length > 1) {
+          addonText = ` <#${id[0]}> and <#${id[1]}> are `;
+          if (id.length === 2) {
+            addonText += `the only two with this channel type.`;
+          } else {
+            id.pop();
```

Comment:
pop shouldnt cause issues because its on a new array (thanks map) but i dont understand why you pop in the first place?

Reply:
Was for when it catches errors, but I just changed the code to do it another way.